pkg/apis/autoscaling/annotation_validation.go

```diff
@@ -145,6 +146,12 @@ func validateMinMaxScale(annotations map[string]string) *apis.FieldError {
 			Paths:   []string{MaxScaleAnnotationKey, MinScaleAnnotationKey},
 		})
 	}
+	if config.MaxScaleLimit > 0 && (max == 0 || max > int64(config.MaxScaleLimit)) {
+		errs = errs.Also(&apis.FieldError{
```

Comment:
This should only fail if we're in create to ensure existing revisions are not affected.